collector/common/conn_pool_stats_test.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2017 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/percona/mongodb_exporter/testutils"
+)
+
+func TestGetConnPoolStatsDecodesFine(t *testing.T) {
+	// setup
+	t.Parallel()
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	t.Run("mongod", func(t *testing.T) {
+		// setup
+		defaultClient := testutils.MustGetConnectedMongodClient(ctx, t)
+		defer defaultClient.Disconnect(ctx)
+		// run
+		statusDefault := GetConnPoolStats(defaultClient)
+		// test
+		assert.NotNil(t, statusDefault)
```

Comment:
Can we check at least _some_ values? This test (and others) will happily pass if an empty structure is returned.

Reply:
Probably yes, but for this: 

1. We should force MongoDB to increase that numbers, as at least in the local environment all numbers returned by this command have `0` value.
2. We should change the test name.

Reply:
Let's do that.

Reply:
> We should force MongoDB to increase that numbers,

I've read the docs and to do that we should configure MongoDB in `replSet` or `cluster` mode, and connect together 2 or more servers. I think it some kind of overengineering for our testing environment at least in this case.

> This test (and others) will happily pass if an empty structure is returned.

To handle this case I've found than MongoDB [has a special field](https://docs.mongodb.com/manual/tutorial/use-database-commands/#command-responses) called `ok` for all responses that were succeeded. I've added this field in our tests. Now those tests wouldn't pass if an empty struct returned.

Reply:
But why they all have 0 value?

Reply:
Because you need to have more than 1 server. Those metrics related to `replSet` or `cluster` mode, so to have connection pool you should have configured cluster first. 😄